File: OLSource1/v1/VS_officedev/codesnippet/VisualBasic/walkthrough--creating-shortcut-menus-for-bookmarks_5.vb

    Private Sub Bookmark_BeforeRightClick(ByVal sender As Object, ByVal e _
    As Microsoft.Office.Tools.Word.ClickEventArgs) _
    Handles Bookmark1.BeforeRightClick, Bookmark2.BeforeRightClick
        Dim startPosition As Integer = 0
        Dim i As Integer

        ' If bookmarks overlap, get bookmark closest to cursor.
        For i = 1 To e.Selection.Bookmarks.Count
            If e.Selection.Bookmarks(i).Start > startPosition Then
                startPosition = e.Selection.Bookmarks(i).Start
            End If
        Next

        ' If closest bookmark is the sender, show the shortcut menu options.
        If DirectCast(sender, Microsoft.Office.Tools.Word.Bookmark).Start = _
            startPosition Then
            selectedBookmark = DirectCast(sender, Microsoft.Office.Tools.Word.Bookmark)

            ShowBoldButton = True
            ShowItalicButton = True

        End If

    End Sub

File: OLSource1/v1/VS_officedev/codesnippet/CSharp/walkthrough--creating-shortcut-menus-for-bookmarks_5.cs

        void bookmark_BeforeRightClick(object sender, ClickEventArgs e)
        {
            int startPosition = 0;

            // If bookmarks overlap, get bookmark closest to cursor.
            for (int i = 1; i <= e.Selection.Bookmarks.Count; i++)
            {
                if (e.Selection.Bookmarks[i].Start > startPosition)
                {
                    startPosition = e.Selection.Bookmarks[i].Start;
                }
            }

            // If closest bookmark is the sender, show shortcut menu options.
            if (((Microsoft.Office.Tools.Word.Bookmark)sender).Start == startPosition)
            {
                selectedBookmark = (Microsoft.Office.Tools.Word.Bookmark)sender;
                
                showBoldButton = true;
                showItalicButton = true;

            }

        }
